oxidation/libparsec/crates/crypto/tests/sequester.rs

```diff
@@ -0,0 +1,285 @@
+RIJnZaPBBhY31yc1YF3blk+Nq5sRhCggEcx7eUStBI0rH5i4OLkKr1ZRHJVLTeB0
+eOup1Orap3/jy42CTVUEFYda0TNuzfe4H0cbjlcJG+rQDCBVMrVkckHfQhRMzKSy
+XmxXmXP3VkO+oDl4QWO9S+f+zvpvMboMCKXsfKr2qPmlYB2kKhjRyTfggWcrNvxs
+zDv+sqW24cup6/zkpJ73NNCO14kA8p6u+33ImTbqjyGu2WvjdeKyK1LMimwhHPER
+9LJ9sHHYnlVxz3bxd2WrBWw+bTSTh6P/0qZTixMIXWABrOzRHIxDpl5LJPLdIAiX
+MQIDAQAB
+-----END PUBLIC KEY-----
+"#;
+
+const PUBLIC_KEY_DER_2048: [u8; 294] = hex!(
+    "30820122300d06092a864886f70d01010105000382010f003082010a0282010100a70d5b19"
+    "19aee2bb8e44d9dfbe26ea44826765a3c1061637d72735605ddb964f8dab9b1184282011cc"
+    "7b7944ad048d2b1f98b838b90aaf56511c954b4de07478eba9d4eadaa77fe3cb8d824d5504"
+    "15875ad1336ecdf7b81f471b8e57091bead00c205532b5647241df42144ccca4b25e6c5799"
+    "73f75643bea039784163bd4be7fecefa6f31ba0c08a5ec7caaf6a8f9a5601da42a18d1c937"
+    "e081672b36fc6ccc3bfeb2a5b6e1cba9ebfce4a49ef734d08ed78900f29eaefb7dc89936ea"
+    "8f21aed96be375e2b22b52cc8a6c211cf111f4b27db071d89e5571cf76f17765ab056c3e6d"
+    "349387a3ffd2a6538b13085d6001acecd11c8c43a65e4b24f2dd200897310203010001");
+
+#[test]
+fn only_rsa_is_supported() {
+    let unsupported_key = hex!(
+        "308201b73082012b06072a8648ce3804013082011e02818100f8df308877a3bcf66db78734"
+        "882f854f934cb5e3bc3cd284b62751d2b6490fd0793dfd43e68393c5fe02f897236fe40949"
+        "892aa7718939be9aec22e813c7ab55a245b0b3d2cef69123c862c2449af305ce30784929a3"
+        "ba4ac3e5f14cf50508558ee9600c74b5ff1020fe3728ff92c303292dcf13c54aac726ba462"
+        "bfee65d3b302150086d038ac40999b7fb3cb7d6253f9ff71a932ca57028180517dea95129a"
+        "3e46e7aee51e00fd4eb0bc90eec8c340fe27dc8e72287ebddbdb7c748b67d7416a54b12a1b"
+        "d09248d1b6e9291a7d266c02fcf76c887a710065e0fdc767e74dfa13edb5d1a8bca331dd32"
+        "dccc199cc2055b446b35a30bd6edde35d08cfbbfd2fbbda2fecce75d4ab5eab2c772cbf914"
+        "1325e0e6e05348b035b3ff03818500028181008fb6c7096b0f95aa2baa71e32ba1c1f54088"
+        "59f573a4786c55fe050e29180a60b6487885c18a294d1f4655032c1fc3c88eb113dae0cddb"
+        "cd5faa40a312dc339256e019aeffd2410c71ef00bf2c519444f3d43024adfc55be6e5db2b7"
+        "8e7d6c54c6cc882ffa606e76e53fc73b3443347507e91227944aeed42004a305de09918b");
+
+    assert!(SequesterPublicKeyDer::try_from(&unsupported_key[..]).is_err());
+    assert!(SequesterVerifyKeyDer::try_from(&unsupported_key[..]).is_err());
+}
+
+#[rstest]
+#[case(&PUBLIC_KEY_DER_1024[..])]
+#[case(&PUBLIC_KEY_DER_2048[..])]
+fn test_key_equality(#[case] pub_key: &[u8]) {
+    let pub_key = SequesterPublicKeyDer::try_from(pub_key).unwrap();
+
+    let pub_key2 = pub_key.clone();
+
+    assert_eq!(pub_key, pub_key2);
+}
+
+#[rstest]
+#[case(1024)]
+#[case(2048)]
+fn test_size(#[case] size_in_bits: usize) {
+    let (priv_key, pub_key) = SequesterPrivateKeyDer::generate_pair(size_in_bits);
+    let (signing_key, verify_key) = SequesterSigningKeyDer::generate_pair(size_in_bits);
+    assert_eq!(priv_key.size_in_bytes(), size_in_bits / 8);
```

Comment:
@FirelightFlagboy

Reply:
Ok we test the `size_in_bytes()` from `impl SequesterPrivateKeyDer` but the one from `impl EnforceDeserialize for SequesterPrivateKeyDer` ?

Reply:
It's the same ?

Reply:
No it's not :)

Reply:
see https://github.com/Scille/parsec-cloud/pull/3629#discussion_r1046774184

Reply:
Well, If it was recursive, the encryption wouldn't work